### PR TITLE
feat(netlify): more resilient docs alert implementation

### PR DESCRIPTION
**Pull request by bpas247**
Continuation of #5078.

Uses the already built-in netlify configuration variables that
will be injected by the netlify build environment, which should
allow use to not have to configure anything extra to get this
feature to work properly.

Also adds some more information about the deployment, specifying
whether or not it's a pull request (and redirects to the right
pr link), or if it's being deployed from a specific branch.

### CODE REVIEW

**Comment by taion** [comment]
wow, gatsby is magic?

**Comment by bpas247** [comment]
> wow, gatsby is magic?

more like netlify is magic 😛 there are [build env variables](https://docs.netlify.com/configure-builds/environment-variables/#netlify-configuration-variables) that get injected when the docs are deployed.

I'm not entirely sure why the docs haven't auto-deployed though, I might trigger a manual deployment for this PR to ensure that everything works properly.

**Comment by taion** [comment]
right, but i mean, gatsby would have to be smart enough to inject them into the static build so they're accessible there...

**Comment by bpas247** [comment]
> right, but i mean, gatsby would have to be smart enough to inject them into the static build so they're accessible there...

yeah I'm kinda worried that gatsby might not be smart enough, but once I figure out why it isn't deploying this branch it should be easy to enough to confirm.

**Comment by jquense** [comment]
There are also some Gatsby netlify plugins might be helpful idk

**Comment by bpas247** [comment]
@jquense do you happen to know why deploy previews aren't being built for PRs? seems like something changed after March 29th, since deploy previews for PRs stopped building since then.

**Comment by bpas247** [comment]
after further investigation, it looks like [this deployment](https://app.netlify.com/sites/react-bootstrap/deploys/5ed5ec09cde94a3190719c3f) (which was a manual deployment) somehow caused netlify's CD to stop working *entirely* (related to #5297), as there hasn't been another deployment since then.

I'm not entirely sure why though 🤔 might have to do with netlify's cache being cleared.

**Comment by bpas247** [comment]
Update: I ended up opening a new issue thread on the netlify community forums to track what's going on with our CD and live preview deployment issues.
https://community.netlify.com/t/cd-and-live-previews-has-suddenly-stopped-working/19344

**Comment by bpas247** [comment]
Update: Netlify issues have been resolved and this PR is ready for review 🚢

**Comment by bpas247** [comment]
Should be ready for review now :shipit: ended up just injecting the build-time variables into the static build

File: www/src/components/MasterDocsAlert.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            href={`https://github.com/react-bootstrap/react-bootstrap/pull/${netlify.reviewId}`}
+          >
+            pull request
+          </Alert.Link>
+        </span>
+      );
+    } else {
+      deploymentDetails = `the ${netlify.branch} branch`;
+    }
+  }
```
Comment (taion) [inline www/src/components/MasterDocsAlert.js:24]: **nitpick:** inline this?

File: www/src/components/MasterDocsAlert.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        <Alert variant="warning" className="w-100">
+          You are currently viewing the auto-generated docs from{' '}
+          {deploymentDetails}. The docs for the current release are available at{' '}
+          <Alert.Link href="https://react-bootstrap.github.io/">
+            https://react-bootstrap.github.io/
+          </Alert.Link>
+        </Alert>
+      </Row>
+    </Container>
+  ) : null;
```
Comment (taion) [inline www/src/components/MasterDocsAlert.js:37]: **nitpick:** no need for ternary here

```suggestion
  return netlify && (
    <Container fluid>
      <Row>
        <Alert variant="warning" className="w-100">
          You are currently viewing the auto-generated docs from{' '}
          {deploymentDetails}. The docs for the current release are available at{' '}
          <Alert.Link href="https://react-bootstrap.github.io/">
            https://react-bootstrap.github.io/
          </Alert.Link>
        </Alert>
      </Row>
    </Container>
  );
```

File: www/src/components/MasterDocsAlert.js (hunk 1 above)
Comment (taion) [reply www/src/components/MasterDocsAlert.js:24]: (without the netlify condition)

**Comment by taion** [comment]
@bpas247 just to check – is it some transient thing here that we have 2 netlify deploy previews for this build?

**Comment by bpas247** [comment]
> @bpas247 just to check – is it some transient thing here that we have 2 netlify deploy previews for this build?

I think somebody created a new [deployment](https://app.netlify.com/sites/laughing-clarke-0d7e9f/overview) since the old deployment wasn't working.

We should delete this deployment since we don't need it anymore. 

I think @jquense is the only person who has perms to delete deployments on netlify, but I can try to disable it in the meantime.

**Comment by bpas247** [comment]
@taion is it cool if I merge? I can't think of anything else I want to do here haha

**Pull request merged**